### PR TITLE
Update GitHub Actions workflow for PR checks

### PR DESCRIPTION
**Pull request by zsmb13**
### Description

Reconfigures our previous PR check jobs into two jobs (lint and test), which are per task, instead of per module. This simplifies our config, as we no longer need to update it when new modules are added.

Workflow execution times should remain roughly the same as before, around 7 minutes.

Removes coverage jobs from CI, as we don't really use them, and they take longer than the others.

### Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] ~Changelog updated with client-facing changes~
- [ ] ~New code is covered by unit tests~
- [ ] ~Comparison screenshots added for visual changes~
- [x] Reviewers added

### CODE REVIEW

File: .github/workflows/pr-checks.yml [hunk 1]
```diff
@@ -38,59 +26,11 @@ jobs:
...
       - name: Unit tests
-        run: ./gradlew :stream-chat-android-ui-components:testDebug :stream-chat-android-ui-components-sample:testDebug --stacktrace
+        run: ./gradlew testDebug --stacktrace
       - name: Upload testDebugUnitTest results
         uses: actions/upload-artifact@v2
         if: failure()
         with:
           name: testDebugUnitTest
-          path: stream-chat-android-ui-components/build/reports/tests/testDebugUnitTest
+          path: /**/build/reports/tests/testDebugUnitTest
```
Comment (JcMinarro) [inline .github/workflows/pr-checks.yml:36]: It will break tests report, right?
I mean, some files will be overridden

Comment (JcMinarro) [reply .github/workflows/pr-checks.yml:36]: Or it will keep folder structure?

Comment (zsmb13) [reply .github/workflows/pr-checks.yml]: I think that should be okay, [the docs for the Action](https://github.com/actions/upload-artifact#upload-using-multiple-paths-and-exclusions) say this (see also the example below this in the docs):

> If a wildcard pattern is used, the path hierarchy will be preserved after the first wildcard pattern.

But I'll break some tests and verify that it works correctly.

Comment (zsmb13) [reply .github/workflows/pr-checks.yml]: This is the structure in the zip produced, I think we're good: 

image

Comment (zsmb13) [reply .github/workflows/pr-checks.yml]: That artifact is here https://github.com/GetStream/stream-chat-android/pull/942/checks?check_run_id=1491606030

**Review by JcMinarro — APPROVED** [review]

**Pull request merged**